src/core/backtrackable_tbl.ml

```diff
@@ -63,6 +64,7 @@ module Make (A : ARG) = struct
   let[@inline] pop_levels self n =
     BS.pop_levels self.undo n ~f:(apply_undo self)
 
+  let[@inline] drop_levels self n = BS.pop_levels self.undo n ~f:ignore
```

Comment:
Feels dangerous to me, because various bits will stay behind in the table after a call to `drop_levels`, when they likely should have been removed. Is it obvious that this is the right thing to do in some use case? Maybe renamed it to something that makes this clearer, e.g. `pop_and_keep`?